packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -104,24 +167,330 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
+        const { changeStart, oldChangeEnd, newChangeEnd } = findStringsDiffIndexes(
+          inputTextValue,
+          newValue,
+          selectionEnd
+        );
+        // get updated html string
+        const change = newValue.substring(changeStart, newChangeEnd);
+        const [updatedHTML, updatedChangeNewEndIndex] = updateHTML(
+          textValue,
+          inputTextValue,
+          newValue,
+          tagsValue,
+          changeStart,
+          oldChangeEnd,
+          change,
+          triggerText
+        );
+        result = updatedHTML;
+        if (updatedChangeNewEndIndex !== null) {
+          if (
+            (change.length === 1 && event.currentTarget.selectionStart === event.currentTarget.selectionEnd) || // simple input
+            (change.length === 0 && newChangeEnd === changeStart) //delete
+          ) {
+            setCaretIndex(updatedChangeNewEndIndex);
+          }
+        }
+      }
+
+      onChange && onChange(event, result);
+    },
+    [
+      onChange,
+      mentionLookupOptions,
+      tagsValue,
+      textValue,
+      inputTextValue,
+      currentTriggerStartIndex,
+      setCaretIndex,
+      setCaretPosition,
+      updateMentionSuggestions,
+      debouncedQueryUpdate,
+      textFieldRef
+    ]
+  );
+
+  /* @conditional-compile-remove(mention) */
+  const updateSelectionIndexesWithMentionIfNeeded = (
+    event: FormEvent<HTMLInputElement | HTMLTextAreaElement>
+  ): void => {
+    let updatedStartIndex = event.currentTarget.selectionStart;
+    let updatedEndIndex = event.currentTarget.selectionEnd;
+    if (
+      event.currentTarget.selectionStart === event.currentTarget.selectionEnd &&
+      event.currentTarget.selectionStart !== null &&
+      event.currentTarget.selectionStart !== -1
+    ) {
+      const mentionTag = findMentionTagForSelection(tagsValue, event.currentTarget.selectionStart);
+      if (mentionTag !== undefined && mentionTag.plainTextBeginIndex !== undefined) {
+        if (selectionStartValue === null) {
+          updatedStartIndex = mentionTag.plainTextBeginIndex;
+          updatedEndIndex = mentionTag.plainTextEndIndex ?? mentionTag.plainTextBeginIndex;
+        } else {
+          const newSelectionIndex = findNewSelectionIndexForMention(
+            mentionTag,
+            inputTextValue,
+            event.currentTarget.selectionStart,
+            selectionStartValue
+          );
+
+          updatedStartIndex = newSelectionIndex;
+          updatedEndIndex = newSelectionIndex;
+        }
+      }
+    } else if (event.currentTarget.selectionStart !== event.currentTarget.selectionEnd) {
+      // Both e.currentTarget.selectionStart !== selectionStartValue and e.currentTarget.selectionEnd !== selectionEndValue can be true when a user selects a text by double click
+      if (event.currentTarget.selectionStart !== null && event.currentTarget.selectionStart !== selectionStartValue) {
```

Comment:
use undefined instead of null
A bit more context:
if you do 
`typeof null`
 in js, the return value will be Object, this will create strange regression when we do the runtime type check

Reply:
The typing of the events are using null, so we've just continued that use internally. We can add to the refactor task to use undefined internally, but I'm curious why null is use for the event typing.